developing-fcc-software/FccSoftwareGit.md

```diff
@@ -119,6 +186,89 @@ Please always follow the recommendations below.
 
     Here, you may write a few more lines if needed
     ```
+### Pull Request description
+
+When opening a new Pull Request, please fill the following information in text box.
+
+1. Please fill a short description what your PR is changing, to be added in the release notes of the repository. This text has the following syntax
+```
+BEGINRELEASENOTES
+- New/Extends XXX
+- Fixes issue XXX
+
+ENDRELEASENOTES
+```
+
+2. In the description, give a short bullet-point list of what was done. If your PR is fixing an open Issue in GitHub, please
+- include the link to it.
+- Make sure you added a test that shows they are actually fixed
+- In the description mention that you fixed it by referring to the issue: "fixes #<issue-id>" (this will automatically close the issue, see also [GitHub help](https://help.github.com/articles/closing-issues-via-commit-messages/))
+
+
+:::{admonition}
+There are number of requisites that will speed up the revision of the Pull request
+* Give a meaningful title that appropriately describes what you did (e.g. Add new calorimeter clustering)
+* Pull requests of work in progress (to make people aware that you are working on a feature) create a PR starting with "[WIP]"
+* Sufficiently documented code (inline, doxygenm, readmes...)
+* Tests cover the modified code
+* The branch is up to date (see [next section](Keeping-your-local-repository-up-to-date) about how to do it)
+* The pull request only contains minimal necesary changes
+
+In case of modifying detector-related code, please check the following as well:
+* Overlap test, [link](https://hep-fcc.github.io/fcc-tutorials/main/full-detector-simulations/Geometry/Geometry.html#overlap-checking)
+* Simulation test, run with a particle gun of your choice, e.g. [link](https://hep-fcc.github.io/fcc-tutorials/main/full-detector-simulations/Geometry/Geometry.html#modify-an-existing-xml-file)
+:::
+
+### Keeping your local repository up to date
+
+It may happen that your fork or development branch becomes outdated with respect to the main branch in the official (`upstream`) repository (1). To update your development branch of your forked repository, please follow these steps:
+
+1. Fetch all changes from the official repository (1)
+```bash
+git fetch upstream
+```
+
+2. Rebase your development branch onto the latest main branch from the upstream repository
+Some repositories have renamed the `master` branch to `main`. Use autocompletion or check on GitHub to confirm.
+
+```bash
+git rebase -i upstream/master
+```
+
+During the rebase process, you may encounter conflicts if you have modified the same file that has already been changed in the main repository. **Do not panic!** Read carefully the terminal messages and follow their instructions.
+
+#### Handling Merge Conflicts
```

Comment:
This can also be done from Github, close to the bottom of the PR page and can be very quick to do if the conflicts are simple - one can also click some options to take either the version in HEAD from the conflict or the introduced one directly (also possible from the command line, but maybe less known).

Reply:
Thanks! Never used myself. I have added the following to the end of the section, do you think it is ok?

:::{admonition}
GitHub web interface allows to update (rebase) and solve simple merge conflicts as well, see [link](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github)
:::